src/platforms/electron/createWindow.js

```diff
@@ -87,6 +87,8 @@ export default appState => {
   window.loadURL(rendererURL + deepLinkingURI);
   setupBarMenu();
 
+  setLanguage(window);
```

Comment:
If this is an async function that happens after `loadURL`, couldn't that create a race condition where some strings are rendered in English before the translation files are loaded?

Reply:
translations seem to happen on the fly. The original code was also Async. 

My main issue that I had was the `index.electron.html` file was loaded before the `CreateWindow.js` file

Reply:
Hmm I may be misunderstanding how this file works then. I have a meeting with @seanyesmunt momentarily, let me check with him and we'll get an answer to you.

Reply:
I think you need to move `setLanguage` before `loadURL` and you will be fine, since you have an `await` on language fetch. By calling it after, I do think there is a chance some translations will fail. You should be able to confirm/test this by arbitrarily delaying the call to `setLanguage` or the duration of the time to download the translation file.

Reply:
Only async funcitons will wait for the function to finish. Putting it before loadURL won't do anything I don't think.

I could put it in `index.js`? That is an async function? I'm not sure that delaying startup is the best option though. I don't think the delay is worth it, but let me know if you disagree!

Reply:
@peterjgrainger it's possible I'm misunderstanding something. 

What I'm fairly confident of is that if `index.html` loads and runs `ui.js` _before_ the JSON translations are returned from lbry.com, then some translations could return in English.

Because you have an `await` in front of the translations download, I thought moving `setLanguage` up would be sufficient. Either way, the key is making sure the language resource is loaded and stored before `ui.js` runs.

This unfortunately does cause a ~100ms load hit, but avoiding it would be a substantial refactor that I don't think we can swallow right now (although if you're itching for more... :grin:).

Reply:
> What I'm fairly confident of is that if index.html loads and runs ui.js before the JSON translations are returned from lbry.com and stored in window.i18n_messages, then some translations could return in English.

Yes agree completely. The translations  definitely will be loaded after the UI starts to render. Looking again at the index it already adds this delay. https://github.com/lbryio/lbry-desktop/blob/005aad6fdb83678c4e2093ee84da9c4a2848d11d/static/index-electron.html#L26. So at least I wouldn't be slowing things down. 

> Because you have an await in front of the translations download, I thought moving setLanguage up would be sufficient.

I'll have to relocate the code somewhere I can block the thread to stop `ui.js` running. I'll have to take a look.

> This unfortunately does cause a ~100ms load hit, but avoiding it would be a substantial refactor that I don't think we can swallow right now

I'll split up the logic so it gets the language from `localStorage` and if it's English then doesn't do the `fetch` (like the current logic)

> (although if you're itching for more... 😁).
Maybe we could speed this up by storing the translations in  `localStorage`? Maybe for the scope of this change I'll keep it simple :)